### ISSUE TITLE
daylight saving time: behavior of insert data is not compatible with mysql 

### ISSUE THREAD

**Issue by zhexuany** [pingcap/tidb#7030]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
set time_zone = 'America/Los_Angeles';
create table t (dt timestamp);
insert into t value('1919-03-30 02:59:59');
```

2. What did you expect to see?

```
ERROR 1292 (22007): Incorrect datetime value: '1919-03-30 02:59:59' for column 'dt' at row 1
```

3. What did you see instead?
It works in tidb. While I am trying select rows from table, all data are zero-valued. 

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
master

**Comment by zhexuany** [comment]
It will be addressed by #6823

**Comment by zhexuany** [comment]
closing
